fastlane/Fastfile

```diff
@@ -345,6 +349,155 @@ lane :build_test_app_and_frameworks do
   )
 end
 
+lane :xcmetrics do |options|
+  next unless is_check_required(sources: sources_matrix[:xcmetrics], force_check: @force_check)
+
+  ['test_output/', 'performance/', "../#{derived_data_path}/Build/Products"].each { |dir| FileUtils.remove_dir(dir, force: true) }
+
+  match_me
+
+  scan(
+    project: xcode_project,
+    scheme: 'StreamChatUITestsApp',
+    testplan: 'Performance',
+    result_bundle: true,
+    derived_data_path: derived_data_path,
+    cloned_source_packages_path: source_packages_path,
+    clean: is_localhost,
+    xcargs: buildcache_xcargs,
+    sdk: 'iphoneos',
+    skip_detect_devices: true,
+    build_for_testing: true
+  )
+
+  firebase_error = ''
+  xcodebuild_output = ''
+  Dir.chdir("../#{derived_data_path}/Build/Products") do
+    begin
+      sh("zip -r MyTests.zip .")
+      sh("gcloud firebase test ios run --test MyTests.zip --timeout 7m --results-dir test_output --device 'model=iphone14pro,version=16.6,orientation=portrait'")
+    rescue StandardError => e
+      UI.error("Test failed on Firebase:\n#{e}")
+      firebase_error = e
+    end
+
+    sh("gsutil cp -r gs://test-lab-af3rt9m4yh360-mqm1zzm767nhc/test_output/iphone14pro-16.6-en-portrait/xcodebuild_output.log xcodebuild_output.log")
+    xcodebuild_output = File.read('xcodebuild_output.log')
+  end
+
+  sh("git clone git@github.com:GetStream/stream-swift-performance-benchmarks.git #{File.dirname(performance_path)}")
+  branch_performance = xcmetrics_log_parser(log: xcodebuild_output)
+  previous_release_branch = "release/#{last_git_tag}"
+  performance_benchmarks = JSON.parse(File.read(performance_path))
+  previous_release_performance = performance_benchmarks[previous_release_branch]
+
+  markdown_table = "## StreamChat XCMetrics\n| `target` | `metric` | `branch` | `#{previous_release_branch}` | `performance` | `status` |\n| - | - | - | - | - | - |\n"
+  ['testMessageListScrollTime', 'testChannelListScrollTime'].each do |test_name|
+    index = 0
+    ['hitches_total_duration', 'duration', 'hitch_time_ratio', 'frame_count', 'frame_rate', 'number_of_hitches'].each do |metric|
+      release_value = previous_release_performance[test_name][metric]['value']
+      branch_value = branch_performance[test_name][metric]['value']
+      value_extension = branch_performance[test_name][metric]['ext']
+
+      max_stddev = 0.1 # Default Xcode Max STDDEV is 10%
+      status_emoji =
+        if branch_value > release_value && branch_value < release_value + (release_value * max_stddev)
+          '🟡' # Warning if a branch is 10% less performant than the previous release
+        elsif branch_value > release_value
+          '🔴' # Failure if a branch is more than 10% less performant than the previous release
+        else
+          '🟢' # Success if a branch is more performant or equals to the previous release
+        end
```

Comment:
Take a closer look at this. Currently, there is no constant baseline. Technically, the previous release is a baseline for every new PR.

If you think we need a constant baseline, it should not be a problem to introduce one.